corehq/apps/geospatial/static/geospatial/js/models.js

```diff
@@ -811,12 +818,197 @@ hqDefine('geospatial/js/models', [
+        self.assignedFilter.subscribe(() => {
+            if (self.assignedFilter() === 'all') {
+                self.filteredCaseData(self.caseData);
+            } else {
+                self.filteredCaseData(self.caseData.filter(function (caseItem) {
+                    if (self.assignedFilter() === 'unassigned') {
+                        return !caseItem.assignedUserId;
+                    }
+                    return caseItem.assignedUserId;
+                }));
+            }
+            self.totalItems(self.filteredCaseData().length);
+            self.goToPage(1);
+        });
+
+        self.isAllChecked = ko.observable(false);
+        self.toggleSelectAll = function (selectAll) {
+            self.isAllChecked(selectAll);
+            for (const caseItem of self.caseDataPage()) {
+                caseItem.isSelected(self.isAllChecked());
+            }
+        };
+
+        self.loadCaseData = function () {
+            const groupData = self.mapModel.caseGroupsIndex;
+            self.caseData = [];
+            for (const item of self.mapModel.caseMapItems()) {
+                const assignedUserId = groupData[item.itemId].assignedUserId;
+                let assignedUsername = '---';
+                let primaryLocName = '---';
+                if (assignedUserId) {
+                    const userData = groupData[assignedUserId].item.itemData;
+                    assignedUsername = userData.name;
+                    primaryLocName = userData.primary_loc_name;
+                }
+                self.caseData.push(
+                    new AssignmentRow(
+                        item.itemData.name, item.itemId, assignedUserId, assignedUsername, primaryLocName, item
+                    )
+                );
+            }
+
+            loadUserData();
+            self.filteredCaseData(self.caseData);
+            self.totalItems(self.filteredCaseData().length);
+        };
+
+        function loadUserData() {
+            self.userData([]);
+            for (const item of self.mapModel.userMapItems()) {
+                self.userData.push(item.getJson());
+            }
+
+            $('#user-assignment-select').select2({
+                placeholder: gettext('No user selected (unassign mode)'),
+                allowClear: true,
+                data: self.userData(),
+            }).val(null).trigger('change');
+        }
+
+        self.goToPage = function (pageNumber) {
+            self.toggleSelectAll(false);
+            self.currentPage(pageNumber);
+        };
+
+        self.assignUserToCases = function () {
+            const selectedUser = self.mapModel.caseGroupsIndex[self.selectedUserId()];
+            for (const caseItem of self.caseDataPage()) {
+                if (!caseItem.isSelected()) {
+                    continue;
+                }
+
+                caseItem.assignedUsername(
+                    (selectedUser) ? selectedUser.item.itemData.name : '---'
```

Comment:
nit: consider using a constant variable for `---`

Reply:
Addressed in [5a7afca](https://github.com/dimagi/commcare-hq/pull/34972/commits/5a7afcaba37bc1955a661d5c17f90dfe787099a2).